server/service/src/invoice/inbound_shipment/update/generate.rs

```diff
@@ -127,6 +138,11 @@ pub(crate) fn generate(
 
 pub fn should_create_batches(invoice: &InvoiceRow, patch: &UpdateInboundShipment) -> bool {
     let existing_status = &invoice.status;
+
+    if is_updating_donor(patch.update_donor_method.clone()) {
+        return true;
+    };
```

Comment:
This part I wasn't sure about.

I've appended this to the should-update-lines logic to make it another OR condition - but it does mean that maybe we are updating all parts of the line when we change the donor.

I am thinking of a use case where if we change the currency rate (for example), but not the status, then later update the donor, it would also make the currency update changes along with donor changes

So maybe this needs to be separated into its own check -> generate -> upsert functionality separate from existing functionality.

But I'm not sure because I don't know why we are waiting for status changes to make line generate updates.